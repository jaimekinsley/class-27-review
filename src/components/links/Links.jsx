import React from 'react';
import PropTypes from 'prop-types';

const Links = ({ links }) => {
  const linkElements = links.map(linkUrl => (
    <li key={linkUrl}>
      <a href={linkUrl}>{linkUrl}</a>
    </li>
  ));

  return (
    <ul>
      {linkElements}
    </ul>
  );
};

Links.propTypes = {
  links: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Links;
