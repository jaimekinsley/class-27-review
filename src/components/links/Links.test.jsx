import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Links from './Links';

describe('Links component', () => {
  afterEach(() => cleanup());
  it('renders Links', () => {
    const { asFragment } = render(
      <Links
        links={['http://test.com', 'test2.com']}
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
