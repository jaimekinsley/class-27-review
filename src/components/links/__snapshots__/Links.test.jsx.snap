// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links component renders Links 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="http://test.com"
      >
        http://test.com
      </a>
    </li>
    <li>
      <a
        href="test2.com"
      >
        test2.com
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
